CMakePresets.json

```diff
@@ -0,0 +1,77 @@
+{
+  "version": 3,
+  "configurePresets": [
+    {
+      "name": "base",
+      "hidden": true,
+      "binaryDir": "${sourceDir}/build/${presetName}",
+      "generator": "Ninja",
+      "cacheVariables": {
+        "CMAKE_C_COMPILER": "clang",
+        "CMAKE_CXX_COMPILER": "clang++"
```

Comment:
For me on ubuntu jammy, these get installed as `clang-16` and `clang++-16`.

Is there a way you suggest allowing multiple versions of clang like this? Should I use update alternatives, alter my PATH or just symlink these? Can I override these somehow? Is that where `CMakeUserPresets.json` comes in?

Reply:
I think that update-alternatives on ubuntu is the preferred method to solve this issue. this is sort of out of scope for cmake--however, we might be able to add a heuristic in the presets there is some ability to have conditionals, but i'm not sure how powerful they are.

Reply:
the values are set in stone after configuration is complete. so you can also leave the names the same, and just do like CC=clang-16 CXX=clang++-16 when you configure the tree and then that will be permanent.